### PR TITLE
Adds a new gamemode, Shadowlings [WIP]

### PR DESCRIPTION
**Pull request by GuppyLaxx**
[Changelogs]: Adds a new gamemode ported from good ol' paradise, shadowlings. They're like nightmares but it's a whole gamemode based around them where they take over the station by controlling people
(edit) : changing this shit to hippie cause i cant code for shit
Also nothing in this pr works so far so dont merge it

:cl: GuppyLaxx
add: Adds shadowlings
add: Adds the gamemode shadowlings
wip: everything in this PR is WIP
imageadd: added a new light alert for shadowpeople/shadowlings
/:cl:

[why]: it's a fun gamemode that will add a whole bunch to the server, stuff like that

### CODE REVIEW

**Comment by GuppyLaxx** [comment]
someone put the do not merge tag on this

File: code/game/gamemodes/shadowling/shadowling.dm [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+	var/list/datum/mind/shadowling_thralls = list()
+	var/list/shadow_objectives = list()
+	var/required_thralls = 15 //How many thralls are needed (hardcoded for now)
+	var/shadowling_ascended = 0 //If at least one shadowling has ascended
+	var/shadowling_dead = 0 //is shadowling kill
+	var/objective_explanation
+
+
+/proc/is_thrall(var/mob/living/M)
+	return istype(M) && M.mind && ticker && ticker.mode && (M.mind in ticker.mode.shadowling_thralls)
```
Comment (deathride58) [inline code/game/gamemodes/shadowling/shadowling.dm:57]: ticker > SSticker

File: code/game/gamemodes/shadowling/shadowling.dm [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+
+
+/proc/is_shadow_or_thrall(var/mob/living/M)
+	return istype(M) && M.mind && ticker && ticker.mode && ((M.mind in ticker.mode.shadowling_thralls) || (M.mind in ticker.mode.shadows))
```
Comment (deathride58) [inline code/game/gamemodes/shadowling/shadowling.dm:61]: ticker > SSticker

File: code/game/gamemodes/shadowling/shadowling.dm [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+
+
+/proc/is_shadow(var/mob/living/M)
+	return istype(M) && M.mind && ticker && ticker.mode && (M.mind in ticker.mode.shadows)
```
Comment (deathride58) [inline code/game/gamemodes/shadowling/shadowling.dm:65]: ticker > SSticker

**Pull request closed without merging**